transaction-impl/src/test/java/com/example/auction/transaction/impl/TransactionServiceImplIntegrationTest.java

```diff
@@ -81,26 +83,47 @@ public void shouldCreateTransactionOnAuctionFinished() {
                     .get(5, SECONDS);
             assertEquals(retrievedTransaction, transactionInfoStarted);
         });
+    }
 
+    @Test(expected = NotFound.class)
+    public void shouldNotCreateTransactionWithNoWinner() throws Throwable {
+        UUID itemIdWithNoWinner = UUID.randomUUID();
+        Item itemWithNoWinner = new Item(itemIdWithNoWinner, creatorId, itemData, 5000, ItemStatus.COMPLETED, Optional.of(Instant.now()), Optional.of(Instant.now()), Optional.empty());
+        ItemEvent.AuctionFinished auctionFinishedWithNoWinner = new ItemEvent.AuctionFinished(itemIdWithNoWinner, itemWithNoWinner);
+        itemProducerStub.send(auctionFinishedWithNoWinner);
+
+        try {
+            transactionService.getTransaction(itemIdWithNoWinner)
+                    .handleRequestHeader(authenticate(creatorId))
+                    .invoke()
+                    .toCompletableFuture()
+                    .get(5, SECONDS);
+        }
+        catch(ExecutionException re) {
+            throw re.getCause();
+        }
+        catch (InterruptedException | TimeoutException e) {
+            throw e;
+        }
     }
 
     @Test
-    @Ignore
     public void shouldSubmitDeliveryDetails() {
-        eventually(new FiniteDuration(10, SECONDS), () ->
+        itemProducerStub.send(auctionFinished);
+
+        eventually(new FiniteDuration(15, SECONDS), () -> {
             transactionService.submitDeliveryDetails(itemId)
                     .handleRequestHeader(authenticate(winnerId))
                     .invoke(deliveryInfo)
                     .toCompletableFuture()
-                    .get(5, SECONDS)
-        );
+                    .get(5, SECONDS);
 
-        eventually(new FiniteDuration(10, SECONDS), () -> {
             TransactionInfo retrievedTransaction = transactionService.getTransaction(itemId)
                     .handleRequestHeader(authenticate(creatorId))
                     .invoke()
                     .toCompletableFuture()
                     .get(5, SECONDS);
+
             assertEquals(retrievedTransaction, transactionInfoWithDelivery);
```

Comment:
I'm not sure this test is correct.
I mean, the test does pass, but it uses `eventually` in a way that's not safe and which we shouldn't promote:
 - eventually is a loop that is executing a block of code with an assertion that _eventually_ becomes successful. The key thing is that the code block is run over and over, which means the code in `transactionService.submitDeliveryDetails...` is invoked several times. IIUC, that code is not idempotent in the `TransactionServiceImpl + TransactionEntity` classes.